fnv.go

```diff
@@ -8,21 +11,16 @@ func newDefaultHasher() Hasher {
 }
 
 type fnv64a struct{}
-
+/*
 const (
 	// offset64 FNVa offset basis. See https://en.wikipedia.org/wiki/Fowler–Noll–Vo_hash_function#FNV-1a_hash
 	offset64 = 14695981039346656037
 	// prime64 FNVa prime value. See https://en.wikipedia.org/wiki/Fowler–Noll–Vo_hash_function#FNV-1a_hash
 	prime64 = 1099511628211
 )
-
+*/
 // Sum64 gets the string and returns its uint64 hash value.
 func (f fnv64a) Sum64(key string) uint64 {
-	var hash uint64 = offset64
-	for i := 0; i < len(key); i++ {
-		hash ^= uint64(key[i])
-		hash *= prime64
-	}
-
+	hash := xxhash.Sum64([]byte(key))
```

Comment:
Please provide a detailed explanation as to why this is more appropriate than FNV hashing, and the problems it solves which the FNV hashing cannot. Looking at the library, it contains this statement: _This is a high-quality hashing algorithm that is much faster than anything in the Go standard library._ We aren't using a golang standard library hashing function, we have implemented our own.

Reply:
1. quality of xxhash is much higher than the rest and faster:
http://cyan4973.github.io/xxHash/
2. https://github.com/cespare/xxhash
i've used it and a few other golang repo also used it and it's actually very good and stable and reliable. the only thing i cant understand is "we only go with standard library... golang" that doesnt really mean anything because if this kind of rationale is needed means: we should all stick to map[][] and not have bigcache... since bigcache is not standard caching of map items etc.
3. xxhash is used in two other very good repo too:
https://github.com/VictoriaMetrics/fastcache
https://github.com/coocood/freecache
4. golang (language version) is going to upgrade anyway, so let's used the best (hash) for now.
means none of us really know whether next gc written's map[][] will improve better than what bigcache is offered now etc etc.

Reply:
Can you run our benchmarks?